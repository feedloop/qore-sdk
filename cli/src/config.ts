import Conf, { Schema } from "conf";

export type CLIConfig = { token: string; project: string; org: string };

export const schema: Schema<CLIConfig> = {
  token: { type: "string", description: "Token" },
  project: { type: "string", description: "Project ID" },
  org: { type: "string", description: "Organization ID" }
};

const config = new Conf({
  schema
});

export default config;
