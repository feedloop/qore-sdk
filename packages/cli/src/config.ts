import Conf, { Schema } from "conf";

export type CLIConfig = { adminSecret: string; url: string };

export const schema: Schema<CLIConfig> = {
  adminSecret: {
    type: "string",
    description: "admin secret",
    default: "admin-secret"
  },
  url: {
    type: "string",
    description: "project url",
    default: "http://localhost:8080/"
  }
};

const config = new Conf({
  schema
});

export default config;
