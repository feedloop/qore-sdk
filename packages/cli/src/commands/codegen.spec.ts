import { setupRecorder } from "nock-record";
import fs from "fs";
import fse from "fs-extra";
import path from "path";
import Codegen from "./codegen";
import config from "../config";

const record = setupRecorder();

describe("codegen", () => {
  beforeEach(() => {
    config.reset("org", "project", "token");
  });
  it("should be able to generate codegen", async () => {
    process.env.QORE_SERVER =
      "https://p-qore-dot-pti-feedloop.et.r.appspot.com";
    const { completeRecording } = await record("codegen");
    await Codegen.run([
      "--project",
      "I0D3NimZQ9GKEDP",
      "--org",
      "lIdfC42DJCN2XzQ",
      "--token",
      "3960f3b8-a139-42eb-8295-3d669e4da4c9",
      "--path",
      "codegen-dest"
    ]);
    completeRecording();
    const qoreGenerated = path.resolve(
      process.cwd(),
      "codegen-dest",
      "qore-env.d.ts"
    );
    expect(
      fse.readFileSync(qoreGenerated, { encoding: "utf-8" })
    ).toMatchSnapshot();
    fs.unlinkSync(qoreGenerated);

    const qoreConfig = path.resolve(
      process.cwd(),
      "codegen-dest",
      "qore.config.json"
    );
    expect(await fse.readJSON(qoreConfig)).toMatchSnapshot();
    fs.unlinkSync(qoreConfig);
    const qoreSchema = path.resolve(
      process.cwd(),
      "codegen-dest",
      "qore.schema.json"
    );
    expect(await fse.readJSON(qoreSchema)).toMatchSnapshot();
    fs.unlinkSync(qoreSchema);
  });
});
