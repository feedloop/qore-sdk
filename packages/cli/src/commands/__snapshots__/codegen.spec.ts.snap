// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`codegen should be able to generate codegen 1`] = `
"// [WARNING] This file is generated by running \`$ qore codegen\` on your root project, please do not edit

/// <reference types=\\"@feedloop/qore-client\\" />
import { QoreSchema } from \\"@feedloop/qore-client\\";

declare module \\"@feedloop/qore-client\\" {
  type MemberTableRow = {
    id: string;
    email: string;
    role: { id: string; displayField: string };
    toDo: { nodes: ToDoTableRow[] };
    totalTask: number;
    totalPoint: number;
    totalHardTask: number;
    taskDone: number;
    password: string;
  };

  type ToDoTableRow = {
    id: string;
    task: string;
    description: string;
    done: boolean;
    points: number;
    deadline: Date;
    difficulty: \\"Easy\\" | \\"Medium\\" | \\"Hard\\";
    person: { nodes: MemberTableRow[] };
    role: { id: string; displayField: string };
    attachment: string;
    timeAllocation: string;
  };

  type DoneViewRow = {
    read: {
      id: string;
      task: string;
      description: string;
      done: boolean;
      points: number;
      deadline: Date;
    };
    write: {
      task: string;
      description: string;
      done: boolean;
      points: number;
      deadline: Date;
    };
    params: {};
    actions: {};
  };

  type MemberDefaultViewViewRow = {
    read: {
      id: string;
      email: string;
      role: { id: string; displayField: string };
      password: string;
    };
    write: {
      email: string;
      password: string;
    };
    params: {};
    actions: {
      createTask: {
        task?: string;
        description?: string;
      };
      addTask: {
        task?: string;
        description?: string;
      };
      insertTask: {};
      createTask2: {
        task?: string;
      };
    };
  };

  type MemberMetricViewRow = {
    read: {
      id: string;
      email: string;
      role: { id: string; displayField: string };
      totalTask: number;
      totalPoint: number;
      totalHardTask: number;
      taskDone: number;
    };
    write: {
      email: string;
    };
    params: {
      email: string;
    };
    actions: {};
  };

  type RamasTaskViewRow = {
    read: {
      id: string;
      task: string;
      description: string;
      done: boolean;
      points: number;
      deadline: Date;
      difficulty: \\"Easy\\" | \\"Medium\\" | \\"Hard\\";
      person: { nodes: MemberTableRow[] };
      role: { id: string; displayField: string };
      attachment: string;
      timeAllocation: string;
    };
    write: {
      task: string;
      description: string;
      done: boolean;
      points: number;
      deadline: Date;
      difficulty: \\"Easy\\" | \\"Medium\\" | \\"Hard\\";
      person: string[];
      attachment: string;
    };
    params: {};
    actions: {};
  };

  type ReisTaskViewRow = {
    read: {
      id: string;
      task: string;
      description: string;
      done: boolean;
      points: number;
      deadline: Date;
      difficulty: \\"Easy\\" | \\"Medium\\" | \\"Hard\\";
      person: { nodes: MemberTableRow[] };
      role: { id: string; displayField: string };
      attachment: string;
      timeAllocation: string;
    };
    write: {
      task: string;
      description: string;
      done: boolean;
      points: number;
      deadline: Date;
      difficulty: \\"Easy\\" | \\"Medium\\" | \\"Hard\\";
      person: string[];
      attachment: string;
    };
    params: {
      timeAllocation?: string;
    };
    actions: {};
  };

  type TestViewRow = {
    read: {
      id: string;
      task: string;
      description: string;
    };
    write: {
      task: string;
      description: string;
    };
    params: {};
    actions: {};
  };

  type ToDoViewRow = {
    read: {
      id: string;
      task: string;
      description: string;
      done: boolean;
      points: number;
      deadline: Date;
    };
    write: {
      task: string;
      description: string;
      done: boolean;
      points: number;
      deadline: Date;
    };
    params: {};
    actions: {};
  };

  type ToDoDefaultViewViewRow = {
    read: {
      id: string;
      task: string;
      description: string;
      done: boolean;
      points: number;
      deadline: Date;
      difficulty: \\"Easy\\" | \\"Medium\\" | \\"Hard\\";
      person: { nodes: MemberTableRow[] };
      role: { id: string; displayField: string };
      timeAllocation: string;
      attachment: string;
    };
    write: {
      task: string;
      description: string;
      done: boolean;
      points: number;
      deadline: Date;
      difficulty: \\"Easy\\" | \\"Medium\\" | \\"Hard\\";
      person: string[];
      attachment: string;
    };
    params: {
      \\"$by.points\\"?: \\"desc\\";
    };
    actions: {};
  };

  type UndoneViewRow = {
    read: {
      id: string;
      task: string;
      description: string;
      done: boolean;
      points: number;
      deadline: Date;
      difficulty: \\"Easy\\" | \\"Medium\\" | \\"Hard\\";
      person: { nodes: MemberTableRow[] };
      role: { id: string; displayField: string };
      attachment: string;
      timeAllocation: string;
    };
    write: {
      task: string;
      description: string;
      done: boolean;
      points: number;
      deadline: Date;
      difficulty: \\"Easy\\" | \\"Medium\\" | \\"Hard\\";
      person: string[];
      attachment: string;
    };
    params: {
      undone?: string;
    };
    actions: {};
  };

  interface ProjectSchema extends QoreSchema {
    done: DoneViewRow;
    memberDefaultView: MemberDefaultViewViewRow;
    memberMetric: MemberMetricViewRow;
    ramasTask: RamasTaskViewRow;
    reisTask: ReisTaskViewRow;
    test: TestViewRow;
    toDo: ToDoViewRow;
    toDoDefaultView: ToDoDefaultViewViewRow;
    undone: UndoneViewRow;
  }
}
"
`;

exports[`codegen should be able to generate codegen 2`] = `
Object {
  "WARNING": "[WARNING] This file is generated by running \`$ qore codegen\` on your root project, please do not edit",
  "authenticationId": "53dDS3FsAHJhpJr",
  "endpoint": "https://stg-qore-app.qorebase.io",
  "organizationId": "lIdfC42DJCN2XzQ",
  "projectId": "I0D3NimZQ9GKEDP",
  "version": "v1",
}
`;

exports[`codegen should be able to generate codegen 3`] = `
Object {
  "WARNING": "[WARNING] This file is generated by running \`$ qore codegen\` on your root project, please do not edit",
  "forms": Array [
    Object {
      "fields": Array [
        Object {
          "id": "task",
        },
        Object {
          "id": "description",
        },
        Object {
          "id": "done",
        },
        Object {
          "id": "points",
        },
        Object {
          "id": "deadline",
        },
      ],
      "id": "newForm3MuYd8OVj6FdlgEnmUv",
      "name": "new form 3muYd8OVj-6fdlg-EnmUv",
      "tableId": "toDo",
    },
  ],
  "roles": Array [
    Object {
      "id": "6Hc0YGiQGoc3Huf",
      "isAdmin": true,
      "name": "admin",
    },
    Object {
      "id": "1gjCDJq3oX0I6RZ",
      "isAdmin": false,
      "name": "Fullstack",
    },
  ],
  "tables": Array [
    Object {
      "fields": Array [
        Object {
          "deletionProtection": true,
          "displayField": true,
          "id": "email",
          "name": "email",
          "type": "text",
        },
        Object {
          "deletionProtection": true,
          "displayField": false,
          "id": "role",
          "name": "role",
          "type": "role",
        },
        Object {
          "column": "person",
          "deletionProtection": false,
          "displayField": false,
          "id": "toDo",
          "multiple": true,
          "name": "To Do",
          "table": "toDo",
          "type": "relation",
        },
        Object {
          "aggregate": "count",
          "condition": null,
          "deletionProtection": false,
          "destinations": Array [
            "id",
          ],
          "displayField": false,
          "id": "totalTask",
          "name": "Total task",
          "source": "toDo",
          "type": "rollup",
        },
        Object {
          "aggregate": "sum",
          "condition": null,
          "deletionProtection": false,
          "destinations": Array [
            "points",
          ],
          "displayField": false,
          "id": "totalPoint",
          "name": "Total point",
          "source": "toDo",
          "type": "rollup",
        },
        Object {
          "aggregate": "count",
          "condition": "difficulty = 'Hard'",
          "deletionProtection": false,
          "destinations": Array [
            "id",
          ],
          "displayField": false,
          "id": "totalHardTask",
          "name": "Total hard task",
          "source": "toDo",
          "type": "rollup",
        },
        Object {
          "condition": null,
          "deletionProtection": false,
          "displayField": false,
          "id": "createTask",
          "name": "Create task",
          "parameters": Array [
            Object {
              "slug": "task",
              "type": "text",
            },
            Object {
              "slug": "description",
              "type": "text",
            },
          ],
          "tasks": Array [
            Object {
              "insert": Object {
                "description": "$param.description",
                "person": "id",
                "task": "$param.task",
              },
              "table": "toDo",
              "type": "insert",
            },
          ],
          "type": "action",
        },
        Object {
          "condition": null,
          "deletionProtection": false,
          "displayField": false,
          "id": "addTask",
          "name": "Add Task",
          "parameters": Array [
            Object {
              "slug": "task",
              "type": "text",
            },
            Object {
              "slug": "description",
              "type": "text",
            },
          ],
          "tasks": Array [
            Object {
              "insert": Object {},
              "table": "toDo",
              "type": "insert",
            },
          ],
          "type": "action",
        },
        Object {
          "aggregate": "count",
          "condition": "done = 'TRUE'",
          "deletionProtection": false,
          "destinations": Array [
            "id",
          ],
          "displayField": false,
          "id": "taskDone",
          "name": "Task Done",
          "source": "toDo",
          "type": "rollup",
        },
        Object {
          "condition": null,
          "deletionProtection": false,
          "displayField": false,
          "id": "insertTask",
          "name": "Insert Task",
          "parameters": Array [],
          "tasks": Array [
            Object {
              "insert": Object {
                "description": "$param.description",
                "done": "'TRUE'",
                "person": "email",
                "task": "$param.task",
              },
              "table": "toDo",
              "type": "insert",
            },
          ],
          "type": "action",
        },
        Object {
          "deletionProtection": false,
          "displayField": false,
          "id": "password",
          "name": "Password",
          "type": "password",
        },
        Object {
          "condition": null,
          "deletionProtection": false,
          "displayField": false,
          "id": "createTask2",
          "name": "CreateTask2",
          "parameters": Array [
            Object {
              "slug": "task",
              "type": "text",
            },
          ],
          "tasks": Array [
            Object {
              "insert": Object {
                "task": "$param.task",
              },
              "table": "toDo",
              "type": "insert",
            },
          ],
          "type": "action",
        },
      ],
      "id": "member",
      "name": "member",
      "type": "auth",
    },
    Object {
      "fields": Array [
        Object {
          "deletionProtection": false,
          "displayField": true,
          "id": "task",
          "name": "Task",
          "type": "text",
        },
        Object {
          "deletionProtection": false,
          "displayField": false,
          "id": "description",
          "name": "description",
          "type": "text",
        },
        Object {
          "deletionProtection": false,
          "displayField": false,
          "id": "done",
          "name": "Done",
          "type": "boolean",
        },
        Object {
          "deletionProtection": false,
          "displayField": false,
          "id": "points",
          "name": "Points",
          "type": "number",
        },
        Object {
          "deletionProtection": false,
          "displayField": false,
          "id": "deadline",
          "name": "Deadline",
          "type": "date",
        },
        Object {
          "deletionProtection": false,
          "displayField": false,
          "id": "difficulty",
          "name": "Difficulty",
          "select": Array [
            "Easy",
            "Medium",
            "Hard",
          ],
          "type": "select",
        },
        Object {
          "column": "toDo",
          "deletionProtection": false,
          "displayField": false,
          "id": "person",
          "multiple": true,
          "name": "Person",
          "table": "member",
          "type": "relation",
        },
        Object {
          "deletionProtection": false,
          "destinations": Array [
            "role",
          ],
          "displayField": false,
          "id": "role",
          "multiple": false,
          "name": "Role",
          "returnType": "table",
          "source": "person",
          "type": "lookup",
        },
        Object {
          "deletionProtection": false,
          "displayField": false,
          "fileType": "image",
          "id": "attachment",
          "name": "Attachment",
          "type": "file",
        },
        Object {
          "deletionProtection": false,
          "displayField": false,
          "formula": "(points < 50) ? 'SHORT' : 'LONG'",
          "id": "timeAllocation",
          "name": "Time Allocation",
          "returnType": "text",
          "type": "formula",
        },
      ],
      "id": "toDo",
      "name": "To Do",
      "type": null,
    },
  ],
  "views": Array [
    Object {
      "design": null,
      "fields": Array [
        Object {
          "deletionProtection": false,
          "design": null,
          "displayField": true,
          "id": "task",
          "name": "Task",
          "type": "text",
        },
        Object {
          "deletionProtection": false,
          "design": null,
          "displayField": false,
          "id": "description",
          "name": "description",
          "type": "text",
        },
        Object {
          "deletionProtection": false,
          "design": null,
          "displayField": false,
          "id": "done",
          "name": "Done",
          "type": "boolean",
        },
        Object {
          "deletionProtection": false,
          "design": null,
          "displayField": false,
          "id": "points",
          "name": "Points",
          "type": "number",
        },
        Object {
          "deletionProtection": false,
          "design": null,
          "displayField": false,
          "id": "deadline",
          "name": "Deadline",
          "type": "date",
        },
      ],
      "id": "done",
      "name": "Done",
      "parameters": Array [],
      "sorts": Array [],
      "tableId": "toDo",
    },
    Object {
      "design": null,
      "fields": Array [
        Object {
          "deletionProtection": true,
          "design": null,
          "displayField": true,
          "id": "email",
          "name": "email",
          "type": "text",
        },
        Object {
          "deletionProtection": true,
          "design": null,
          "displayField": false,
          "id": "role",
          "name": "role",
          "type": "role",
        },
        Object {
          "condition": null,
          "deletionProtection": false,
          "design": null,
          "displayField": false,
          "id": "createTask",
          "name": "Create task",
          "parameters": Array [
            Object {
              "slug": "task",
              "type": "text",
            },
            Object {
              "slug": "description",
              "type": "text",
            },
          ],
          "tasks": Array [
            Object {
              "insert": Object {
                "description": "$param.description",
                "person": "id",
                "task": "$param.task",
              },
              "table": "toDo",
              "type": "insert",
            },
          ],
          "type": "action",
        },
        Object {
          "condition": null,
          "deletionProtection": false,
          "design": null,
          "displayField": false,
          "id": "addTask",
          "name": "Add Task",
          "parameters": Array [
            Object {
              "slug": "task",
              "type": "text",
            },
            Object {
              "slug": "description",
              "type": "text",
            },
          ],
          "tasks": Array [
            Object {
              "insert": Object {},
              "table": "toDo",
              "type": "insert",
            },
          ],
          "type": "action",
        },
        Object {
          "condition": null,
          "deletionProtection": false,
          "design": null,
          "displayField": false,
          "id": "insertTask",
          "name": "Insert Task",
          "parameters": Array [],
          "tasks": Array [
            Object {
              "insert": Object {
                "description": "$param.description",
                "done": "'TRUE'",
                "person": "email",
                "task": "$param.task",
              },
              "table": "toDo",
              "type": "insert",
            },
          ],
          "type": "action",
        },
        Object {
          "deletionProtection": false,
          "design": null,
          "displayField": false,
          "id": "password",
          "name": "Password",
          "type": "password",
        },
        Object {
          "condition": null,
          "deletionProtection": false,
          "design": null,
          "displayField": false,
          "id": "createTask2",
          "name": "CreateTask2",
          "parameters": Array [
            Object {
              "slug": "task",
              "type": "text",
            },
          ],
          "tasks": Array [
            Object {
              "insert": Object {
                "task": "$param.task",
              },
              "table": "toDo",
              "type": "insert",
            },
          ],
          "type": "action",
        },
      ],
      "id": "memberDefaultView",
      "name": "member Default View",
      "parameters": Array [],
      "sorts": Array [],
      "tableId": "member",
    },
    Object {
      "design": null,
      "fields": Array [
        Object {
          "deletionProtection": true,
          "design": null,
          "displayField": true,
          "id": "email",
          "name": "email",
          "type": "text",
        },
        Object {
          "deletionProtection": true,
          "design": null,
          "displayField": false,
          "id": "role",
          "name": "role",
          "type": "role",
        },
        Object {
          "aggregate": "count",
          "condition": null,
          "deletionProtection": false,
          "design": null,
          "destinations": Array [
            "id",
          ],
          "displayField": false,
          "id": "totalTask",
          "name": "Total task",
          "source": "toDo",
          "type": "rollup",
        },
        Object {
          "aggregate": "sum",
          "condition": null,
          "deletionProtection": false,
          "design": null,
          "destinations": Array [
            "points",
          ],
          "displayField": false,
          "id": "totalPoint",
          "name": "Total point",
          "source": "toDo",
          "type": "rollup",
        },
        Object {
          "aggregate": "count",
          "condition": "difficulty = 'Hard'",
          "deletionProtection": false,
          "design": null,
          "destinations": Array [
            "id",
          ],
          "displayField": false,
          "id": "totalHardTask",
          "name": "Total hard task",
          "source": "toDo",
          "type": "rollup",
        },
        Object {
          "aggregate": "count",
          "condition": "done = 'TRUE'",
          "deletionProtection": false,
          "design": null,
          "destinations": Array [
            "id",
          ],
          "displayField": false,
          "id": "taskDone",
          "name": "Task Done",
          "source": "toDo",
          "type": "rollup",
        },
      ],
      "id": "memberMetric",
      "name": "Member Metric",
      "parameters": Array [
        Object {
          "required": true,
          "slug": "email",
          "type": "text",
        },
      ],
      "sorts": Array [],
      "tableId": "member",
    },
    Object {
      "design": null,
      "fields": Array [
        Object {
          "deletionProtection": false,
          "design": null,
          "displayField": true,
          "id": "task",
          "name": "Task",
          "type": "text",
        },
        Object {
          "deletionProtection": false,
          "design": null,
          "displayField": false,
          "id": "description",
          "name": "description",
          "type": "text",
        },
        Object {
          "deletionProtection": false,
          "design": null,
          "displayField": false,
          "id": "done",
          "name": "Done",
          "type": "boolean",
        },
        Object {
          "deletionProtection": false,
          "design": null,
          "displayField": false,
          "id": "points",
          "name": "Points",
          "type": "number",
        },
        Object {
          "deletionProtection": false,
          "design": null,
          "displayField": false,
          "id": "deadline",
          "name": "Deadline",
          "type": "date",
        },
        Object {
          "deletionProtection": false,
          "design": null,
          "displayField": false,
          "id": "difficulty",
          "name": "Difficulty",
          "select": Array [
            "Easy",
            "Medium",
            "Hard",
          ],
          "type": "select",
        },
        Object {
          "column": "toDo",
          "deletionProtection": false,
          "design": null,
          "displayField": false,
          "id": "person",
          "multiple": true,
          "name": "Person",
          "table": "member",
          "type": "relation",
        },
        Object {
          "deletionProtection": false,
          "design": null,
          "destinations": Array [
            "role",
          ],
          "displayField": false,
          "id": "role",
          "multiple": false,
          "name": "Role",
          "returnType": "table",
          "source": "person",
          "type": "lookup",
        },
        Object {
          "deletionProtection": false,
          "design": null,
          "displayField": false,
          "fileType": "image",
          "id": "attachment",
          "name": "Attachment",
          "type": "file",
        },
        Object {
          "deletionProtection": false,
          "design": null,
          "displayField": false,
          "formula": "(points < 50) ? 'SHORT' : 'LONG'",
          "id": "timeAllocation",
          "name": "Time Allocation",
          "returnType": "text",
          "type": "formula",
        },
      ],
      "id": "ramasTask",
      "name": "Rama's Task",
      "parameters": Array [],
      "sorts": Array [],
      "tableId": "toDo",
    },
    Object {
      "design": null,
      "fields": Array [
        Object {
          "deletionProtection": false,
          "design": null,
          "displayField": true,
          "id": "task",
          "name": "Task",
          "type": "text",
        },
        Object {
          "deletionProtection": false,
          "design": null,
          "displayField": false,
          "id": "description",
          "name": "description",
          "type": "text",
        },
        Object {
          "deletionProtection": false,
          "design": null,
          "displayField": false,
          "id": "done",
          "name": "Done",
          "type": "boolean",
        },
        Object {
          "deletionProtection": false,
          "design": null,
          "displayField": false,
          "id": "points",
          "name": "Points",
          "type": "number",
        },
        Object {
          "deletionProtection": false,
          "design": null,
          "displayField": false,
          "id": "deadline",
          "name": "Deadline",
          "type": "date",
        },
        Object {
          "deletionProtection": false,
          "design": null,
          "displayField": false,
          "id": "difficulty",
          "name": "Difficulty",
          "select": Array [
            "Easy",
            "Medium",
            "Hard",
          ],
          "type": "select",
        },
        Object {
          "column": "toDo",
          "deletionProtection": false,
          "design": null,
          "displayField": false,
          "id": "person",
          "multiple": true,
          "name": "Person",
          "table": "member",
          "type": "relation",
        },
        Object {
          "deletionProtection": false,
          "design": null,
          "destinations": Array [
            "role",
          ],
          "displayField": false,
          "id": "role",
          "multiple": false,
          "name": "Role",
          "returnType": "table",
          "source": "person",
          "type": "lookup",
        },
        Object {
          "deletionProtection": false,
          "design": null,
          "displayField": false,
          "fileType": "image",
          "id": "attachment",
          "name": "Attachment",
          "type": "file",
        },
        Object {
          "deletionProtection": false,
          "design": null,
          "displayField": false,
          "formula": "(points < 50) ? 'SHORT' : 'LONG'",
          "id": "timeAllocation",
          "name": "Time Allocation",
          "returnType": "text",
          "type": "formula",
        },
      ],
      "id": "reisTask",
      "name": "Task by timeAllocation",
      "parameters": Array [
        Object {
          "required": false,
          "slug": "timeAllocation",
          "type": "text",
        },
      ],
      "sorts": Array [],
      "tableId": "toDo",
    },
    Object {
      "design": null,
      "fields": Array [
        Object {
          "deletionProtection": false,
          "design": null,
          "displayField": true,
          "id": "task",
          "name": "Task",
          "type": "text",
        },
        Object {
          "deletionProtection": false,
          "design": null,
          "displayField": false,
          "id": "description",
          "name": "description",
          "type": "text",
        },
      ],
      "id": "test",
      "name": "Priorities",
      "parameters": Array [],
      "sorts": Array [],
      "tableId": "toDo",
    },
    Object {
      "design": null,
      "fields": Array [
        Object {
          "deletionProtection": false,
          "design": null,
          "displayField": true,
          "id": "task",
          "name": "Task",
          "type": "text",
        },
        Object {
          "deletionProtection": false,
          "design": null,
          "displayField": false,
          "id": "description",
          "name": "description",
          "type": "text",
        },
        Object {
          "deletionProtection": false,
          "design": null,
          "displayField": false,
          "id": "done",
          "name": "Done",
          "type": "boolean",
        },
        Object {
          "deletionProtection": false,
          "design": null,
          "displayField": false,
          "id": "points",
          "name": "Points",
          "type": "number",
        },
        Object {
          "deletionProtection": false,
          "design": null,
          "displayField": false,
          "id": "deadline",
          "name": "Deadline",
          "type": "date",
        },
      ],
      "id": "toDo",
      "name": "To Do",
      "parameters": Array [],
      "sorts": Array [],
      "tableId": "toDo",
    },
    Object {
      "design": null,
      "fields": Array [
        Object {
          "deletionProtection": false,
          "design": null,
          "displayField": true,
          "id": "task",
          "name": "Task",
          "type": "text",
        },
        Object {
          "deletionProtection": false,
          "design": null,
          "displayField": false,
          "id": "description",
          "name": "description",
          "type": "text",
        },
        Object {
          "deletionProtection": false,
          "design": null,
          "displayField": false,
          "id": "done",
          "name": "Done",
          "type": "boolean",
        },
        Object {
          "deletionProtection": false,
          "design": null,
          "displayField": false,
          "id": "points",
          "name": "Points",
          "type": "number",
        },
        Object {
          "deletionProtection": false,
          "design": null,
          "displayField": false,
          "id": "deadline",
          "name": "Deadline",
          "type": "date",
        },
        Object {
          "deletionProtection": false,
          "design": null,
          "displayField": false,
          "id": "difficulty",
          "name": "Difficulty",
          "select": Array [
            "Easy",
            "Medium",
            "Hard",
          ],
          "type": "select",
        },
        Object {
          "column": "toDo",
          "deletionProtection": false,
          "design": null,
          "displayField": false,
          "id": "person",
          "multiple": true,
          "name": "Person",
          "table": "member",
          "type": "relation",
        },
        Object {
          "deletionProtection": false,
          "design": null,
          "destinations": Array [
            "role",
          ],
          "displayField": false,
          "id": "role",
          "multiple": false,
          "name": "Role",
          "returnType": "table",
          "source": "person",
          "type": "lookup",
        },
        Object {
          "deletionProtection": false,
          "design": null,
          "displayField": false,
          "formula": "(points < 50) ? 'SHORT' : 'LONG'",
          "id": "timeAllocation",
          "name": "Time Allocation",
          "returnType": "text",
          "type": "formula",
        },
        Object {
          "deletionProtection": false,
          "design": null,
          "displayField": false,
          "fileType": "image",
          "id": "attachment",
          "name": "Attachment",
          "type": "file",
        },
      ],
      "id": "toDoDefaultView",
      "name": "Tasks",
      "parameters": Array [],
      "sorts": Array [
        Object {
          "by": "points",
          "order": "desc",
        },
      ],
      "tableId": "toDo",
    },
    Object {
      "design": null,
      "fields": Array [
        Object {
          "deletionProtection": false,
          "design": null,
          "displayField": true,
          "id": "task",
          "name": "Task",
          "type": "text",
        },
        Object {
          "deletionProtection": false,
          "design": null,
          "displayField": false,
          "id": "description",
          "name": "description",
          "type": "text",
        },
        Object {
          "deletionProtection": false,
          "design": null,
          "displayField": false,
          "id": "done",
          "name": "Done",
          "type": "boolean",
        },
        Object {
          "deletionProtection": false,
          "design": null,
          "displayField": false,
          "id": "points",
          "name": "Points",
          "type": "number",
        },
        Object {
          "deletionProtection": false,
          "design": null,
          "displayField": false,
          "id": "deadline",
          "name": "Deadline",
          "type": "date",
        },
        Object {
          "deletionProtection": false,
          "design": null,
          "displayField": false,
          "id": "difficulty",
          "name": "Difficulty",
          "select": Array [
            "Easy",
            "Medium",
            "Hard",
          ],
          "type": "select",
        },
        Object {
          "column": "toDo",
          "deletionProtection": false,
          "design": null,
          "displayField": false,
          "id": "person",
          "multiple": true,
          "name": "Person",
          "table": "member",
          "type": "relation",
        },
        Object {
          "deletionProtection": false,
          "design": null,
          "destinations": Array [
            "role",
          ],
          "displayField": false,
          "id": "role",
          "multiple": false,
          "name": "Role",
          "returnType": "table",
          "source": "person",
          "type": "lookup",
        },
        Object {
          "deletionProtection": false,
          "design": null,
          "displayField": false,
          "fileType": "image",
          "id": "attachment",
          "name": "Attachment",
          "type": "file",
        },
        Object {
          "deletionProtection": false,
          "design": null,
          "displayField": false,
          "formula": "(points < 50) ? 'SHORT' : 'LONG'",
          "id": "timeAllocation",
          "name": "Time Allocation",
          "returnType": "text",
          "type": "formula",
        },
      ],
      "id": "undone",
      "name": "Undone",
      "parameters": Array [
        Object {
          "required": false,
          "slug": "undone",
          "type": "text",
        },
      ],
      "sorts": Array [],
      "tableId": "toDo",
    },
  ],
}
`;
